Source/WebGPU/WGSL/Metal/MetalFunctionWriter.cpp

```diff
@@ -625,6 +625,38 @@ void FunctionDefinitionWriter::visit(AST::CallExpression& call)
 
     if (is<AST::NamedTypeName>(call.target())) {
         static constexpr std::pair<ComparableASCIILiteral, void(*)(FunctionDefinitionWriter*, AST::CallExpression&)> builtinMappings[] {
+            { "textureLoad", [](FunctionDefinitionWriter* writer, AST::CallExpression& call) {
+                auto& texture = call.arguments()[0];
+                auto& coordinates = call.arguments()[1];
+                writer->stringBuilder().append("({\n");
+                {
+                    IndentationScope scope(writer->indent());
+                    {
+                        writer->stringBuilder().append(writer->indent(), "auto __coords = (");
+                        writer->visit(texture);
+                        writer->stringBuilder().append("_UVRemapMatrix * float3(");
```

Comment:
does this only occur for `texture_external` types? Because we could use textureLoad on a regular texture_2d as well, right?

Reply:
Good catch, the code didn't handle regular textures. I updated the PR